src/main/java/org/broadinstitute/hellbender/utils/codecs/GENCODE/GencodeGtfCodec.java

```diff
@@ -0,0 +1,376 @@
+     * @param leafFeatureStore {@link List} of {@link GencodeGtfFeature}s to insert into corresponding {@link GencodeGtfExonFeature} objects in {@code exonStore}
+     */
+    private void aggregateRecordsIntoGeneFeature(final GencodeGtfGeneFeature gene,
+                                                                  final GencodeGtfTranscriptFeature transcript,
+                                                                  final ArrayList< GencodeGtfExonFeature > exonStore,
+                                                                  final ArrayList< GencodeGtfFeature > leafFeatureStore ) {
+
+        // OK, we go through the record and consolidate the sub parts of the record.
+        // We must consolidate these records through grouping by genomic position.
+
+        // Loop through the Exons and put the correct leaf features into each:
+        for ( GencodeGtfExonFeature exon : exonStore ) {
+            for (Iterator<GencodeGtfFeature> iterator = leafFeatureStore.iterator(); iterator.hasNext(); ) {
+
+                GencodeGtfFeature feature = iterator.next();
+
+                // Features that are within the extents of an exon belong in that exon:
+                if ( exon.contains(feature) ) {
+
+                    GencodeGtfFeature.FeatureType featureType = feature.getFeatureType();
+
+                    // Add the feature to the correct place in the exon:
+                    switch (featureType) {
+                        case cds:
+                            exon.setCds((GencodeGtfCDSFeature) feature);
+                            break;
+                        case start_codon:
+                            exon.setStartCodon((GencodeGtfStartCodonFeature) feature);
+                            break;
+                        case stop_codon:
+                            exon.setStopCodon((GencodeGtfStopCodonFeature) feature);
+                            break;
+                        case utr:
+                            transcript.addUtr((GencodeGtfUTRFeature) feature);
+                            break;
+                        case selenocysteine:
+                            transcript.addSelenocysteine(((GencodeGtfSelenocysteineFeature) feature));
+                            break;
+                        default:
+                            throw new UserException.MalformedFile(
+                                    "Found unexpected Feature Type in GENCODE GTF File (line " +
+                                            feature.getFeatureOrderNumber() + "): " +
+                                            featureType.toString()
+                            );
+                    }
+
+                    // We have used this iterator item.
+                    // We should remove it now so we don't keep going through the list each exon.
+                    iterator.remove();
```

Comment:
I don't think so.  Each feature has an exon_id field that is filled out and expected to be a single value.  I suspect that any features that belong in multiple exons would be duplicated in the data file.